src/prop/cadical.cpp

```diff
@@ -204,6 +204,23 @@ unsigned CadicalSolver::getAssertionLevel() const
 
 bool CadicalSolver::ok() const { return d_inSatMode; }
 
+void CadicalSolver::setDrat()
+{
+  d_dratFile = fopen(tempDratFilePath, "wb");
```

Comment:
Instead of using some fixed name, generate a unique name here so that you do not have have clashes if one were to use differente SAT solvers producing DRAT proofs.

Reply:
like using a unique id or something? is there an example or a standard for this in the code base?

Reply:
Using `getpid` appears to be a reasonable way of doing this. So you could do:

```suggestion
  sprintf(d_tempDratFilePath, "dratProof_%d.drat", getpid());
  d_dratFile = fopen(d_tempDratFilePath, "wb");
```